stdlib/public/runtime/SwiftObject.mm

```diff
@@ -1382,6 +1383,24 @@ static bool usesNativeSwiftReferenceCounting_nonNull(
          swift_isUniquelyReferencedOrPinned_nonNull_native(object);
 }
 
+// Given a non-@objc object reference, return true iff the
+// object is non-nil and has a strong reference count greather than 1
+bool swift::swift_isEscapingClosureAtFileLocation(const HeapObject *object,
+                                                  const unsigned char *filename,
+                                                  int32_t filenameLength,
+                                                  int32_t line) {
+  bool isEscaping =
+      object != nullptr && !object->refCounts.isUniquelyReferenced();
+  if (isEscaping) {
+    auto *fatalErr = reinterpret_cast<const unsigned char *>("Fatal error");
+    auto *message = reinterpret_cast<const unsigned char *>(
+        "closure argument was escaped in withoutActuallyEscaping block");
+    _swift_stdlib_reportFatalErrorInFile(fatalErr, 11, message, 62, filename,
```

Comment:
Please use `strlen` here (LLVM should be able to fold it away).

Aside: maybe we should rename this callback to not have "fatal" in it, since it doesn't abort itself.